=== modified file 'src/Backend/TouchpadSettings.vala'
--- src/Backend/TouchpadSettings.vala	2016-06-18 18:12:17 +0000
+++ src/Backend/TouchpadSettings.vala	2016-08-20 02:12:02 +0000
@@ -22,7 +22,7 @@
     public string click_method { get; set; }
     public double speed { get; set; }
     public string left_handed  { get; set; }
-    public string scroll_method { get; set; }
+    public bool edge_scrolling_enabled { get; set; }
     public bool natural_scroll { get; set; }
 
     public TouchpadSettings () {

=== modified file 'src/Widgets/TouchpadSection.vala'
--- src/Widgets/TouchpadSection.vala	2016-06-18 18:42:34 +0000
+++ src/Widgets/TouchpadSection.vala	2016-08-20 02:23:09 +0000
@@ -24,7 +24,7 @@
     private Gtk.Switch tap_to_click_switch;
     private Gtk.ComboBoxText click_method_combobox;
     private Gtk.Scale pointer_speed_scale;
-    private Gtk.ComboBoxText scrolling_combobox;
+    private Gtk.Switch scrolling_switch;
     private Gtk.Switch horizontal_scrolling_switch;
     private Gtk.Switch natural_scrolling_switch;
 
@@ -56,10 +56,8 @@
         pointer_speed_scale.draw_value = false;
         pointer_speed_scale.set_size_request (160, -1);
 
-        scrolling_combobox = new Gtk.ComboBoxText ();
-        scrolling_combobox.append ("two-finger-scrolling", _("Two-finger"));
-        scrolling_combobox.append ("edge-scrolling", _("Edge"));
-        scrolling_combobox.append ("disabled", _("Disabled"));
+	scrolling_switch = new Gtk.Switch ();
+        scrolling_switch.halign = Gtk.Align.START;
 
         horizontal_scrolling_switch = new Gtk.Switch ();
         horizontal_scrolling_switch.halign = Gtk.Align.START;
@@ -71,15 +69,15 @@
         //this.add_entry (_("Disable while typing:"), disable_while_typing_switch);
         this.add_entry (_("Tap to click:"), tap_to_click_switch);
         this.add_entry (_("Physical clicking:"), click_method_combobox);
-        this.add_entry (_("Scrolling:"), scrolling_combobox);
+        this.add_entry (_("Edge Scrolling:"), scrolling_switch);
         //this.add_entry (_("Horizontal scrolling:"), horizontal_scrolling_switch);
         this.add_entry (_("Natural scrolling:"), natural_scrolling_switch);
     }
 
     private void create_bindings () {
-        touchpad_settings.bind_property ("scroll-method",
-                                         scrolling_combobox,
-                                         "active-id",
+        touchpad_settings.bind_property ("edge-scrolling-enabled",
+                                         scrolling_switch,
+                                         "state",
                                          BindingFlags.BIDIRECTIONAL | BindingFlags.SYNC_CREATE);
 
         touchpad_settings.bind_property ("tap-to-click",

