diff --git a/src/Plug.vala b/src/Plug.vala
index df2df85..2247bf1 100644
--- a/src/Plug.vala
+++ b/src/Plug.vala
@@ -100,6 +100,7 @@ public class About.Plug : Switchboard.Plug {
                         bugtracker_url = bugtracker_url.replace ("\"", "");
                     }
                 }
+                support_url = "https://ask.fedoraproject.org/";
                 if ("SUPPORT_URL=" in line) {
                     support_url = line.replace ("SUPPORT_URL=", "");
                     if ("\"" in support_url) {
@@ -108,25 +109,25 @@ public class About.Plug : Switchboard.Plug {
                 }
             }
         } catch (Error e) {
-            warning("Couldn't read os-release file, assuming elementary OS");
-            os = "elementary OS";
-            website_url = "https://elementary.io";
-            bugtracker_url = "https://bugs.launchpad.net/elementaryos/+filebug";
-            support_url = "https://elementary.io/support";
+            warning("Couldn't read os-release file, assuming Fedora");
+            os = "Fedora";
+            website_url = "https://fedoraproject.org/";
+            bugtracker_url = "https://bugzilla.redhat.com/";
+            support_url = "https://ask.fedoraproject.org/";
 
         }
 
         //Upstream distro version (for "Built on" text)
         //FIXME: Add distro specific field to /etc/os-release and use that instead
         // Like "ELEMENTARYOS_UPSTREAM_PRETTY_ID" or something
-        file = File.new_for_path("/etc/upstream-release/lsb-release");
+        file = File.new_for_path("/etc/os-release");
         try {
             var dis = new DataInputStream (file.read ());
             string line;
             // Read lines until end of file (null) is reached
             while ((line = dis.read_line (null)) != null) {
-                if ("DISTRIB_DESCRIPTION=" in line) {
-                    upstream_release = line.replace ("DISTRIB_DESCRIPTION=", "");
+                if ("PRETTY_NAME=" in line) {
+                    upstream_release = line.replace ("PRETTY_NAME=", "");
                 }
             }
         } catch (Error e) {
@@ -240,7 +241,7 @@ public class About.Plug : Switchboard.Plug {
         main_grid = new Gtk.EventBox ();
 
         // Create the section about elementary OS
-        var logo = new Gtk.Image.from_icon_name ("distributor-logo", Gtk.icon_size_register ("LOGO", 128, 128));
+        var logo = new Gtk.Image.from_icon_name ("fedora-logo-icon", Gtk.icon_size_register ("LOGO", 128, 128));
         logo.halign = Gtk.Align.END;
 
         var title = new Gtk.Label (null);
