diff --git a/src/SettingsDaemon.vala b/src/SettingsDaemon.vala
index 84d62c7..d3fff2e 100644
--- a/src/SettingsDaemon.vala
+++ b/src/SettingsDaemon.vala
@@ -22,37 +22,6 @@ public class SettingsDaemon : Object {
     private int n_names = 0;
 
     public void start () {
-        string[] disabled = { "org.gnome.settings-daemon.plugins.background",
-                              "org.gnome.settings-daemon.plugins.clipboard",
-                              "org.gnome.settings-daemon.plugins.font",
-                              "org.gnome.settings-daemon.plugins.gconf",
-                              "org.gnome.settings-daemon.plugins.gsdwacom",
-                              "org.gnome.settings-daemon.plugins.housekeeping",
-                              "org.gnome.settings-daemon.plugins.keybindings",
-                              "org.gnome.settings-daemon.plugins.keyboard",
-                              "org.gnome.settings-daemon.plugins.media-keys",
-                              "org.gnome.settings-daemon.plugins.mouse",
-                              "org.gnome.settings-daemon.plugins.print-notifications",
-                              "org.gnome.settings-daemon.plugins.smartcard",
-                              "org.gnome.settings-daemon.plugins.sound",
-                              "org.gnome.settings-daemon.plugins.wacom" };
-
-        string[] enabled =  { "org.gnome.settings-daemon.plugins.a11y-keyboard",
-                              "org.gnome.settings-daemon.plugins.a11y-settings",
-                              "org.gnome.settings-daemon.plugins.color",
-                              "org.gnome.settings-daemon.plugins.cursor",
-                              "org.gnome.settings-daemon.plugins.power",
-                              "org.gnome.settings-daemon.plugins.xrandr",
-                              "org.gnome.settings-daemon.plugins.xsettings" };
-
-        foreach (var schema in disabled) {
-            set_plugin_enabled (schema, false);
-        }
-
-        foreach (var schema in enabled) {
-            set_plugin_enabled (schema, true);
-        }
-
         /* Pretend to be GNOME session */
         session_manager = new SessionManagerInterface ();
         n_names++;
@@ -71,28 +40,13 @@ public class SettingsDaemon : Object {
                            () => debug ("Failed to acquire name org.gnome.SessionManager"));
     }
 
-    private void set_plugin_enabled (string schema_name, bool enabled) {
-        var source = SettingsSchemaSource.get_default ();
-        var schema = source.lookup (schema_name, false);
-        if (schema != null) {
-            var settings = new Settings (schema_name);
-            settings.set_boolean ("active", enabled);
-        }
-    }
-
     private void start_settings_daemon () {
         n_names--;
         if (n_names != 0) {
             return;
         }
 
-        debug ("All bus names acquired, starting gnome-settings-daemon");
-
-        try {
-            Process.spawn_command_line_async ("gnome-settings-daemon");
-        } catch (SpawnError e) {
-            debug ("Could not start gnome-settings-daemon: %s", e.message);
-        }
+        return;
     }
 }
 
