diff --git a/src/Plug.vala b/src/Plug.vala
index 5d85ff3..54e97ed 100644
--- a/src/Plug.vala
+++ b/src/Plug.vala
@@ -25,7 +25,6 @@ namespace Power {
     public class Plug : Switchboard.Plug {
         private Gtk.SizeGroup label_size;
         private Gtk.StackSwitcher stack_switcher;
-        private GLib.Settings pantheon_dpms_settings;
 
         private PowerSettings screen;
         private Battery battery;
@@ -41,7 +40,6 @@ namespace Power {
 
         construct {
             settings = new GLib.Settings ("org.gnome.settings-daemon.plugins.power");
-            pantheon_dpms_settings = new GLib.Settings ("org.pantheon.dpms");
 
             battery = new Battery ();
             power_supply = new PowerSupply ();
@@ -298,21 +296,11 @@ namespace Power {
                 main_grid.attach (lock_image, 2, 6, 1, 1);
             }
 
-            var screen_timeout_label = new Gtk.Label (_("Turn off display when inactive for:"));
-            screen_timeout_label.halign = Gtk.Align.END;
-            screen_timeout_label.xalign = 1.0f;
-            label_size.add_widget (screen_timeout_label);
-
-            var screen_timeout = new TimeoutComboBox (pantheon_dpms_settings, "standby-time");
-            screen_timeout.changed.connect (run_dpms_helper);
-
             var power_combobox = new ActionComboBox (_("Power button:"), "power-button-action");
             label_size.add_widget (power_combobox.label);
 
-            main_grid.attach (screen_timeout_label, 0, 3, 1, 1);
-            main_grid.attach (screen_timeout, 1, 3, 1, 1);
-            main_grid.attach (power_combobox.label, 0, 4, 1, 1);
-            main_grid.attach (power_combobox, 1, 4, 1, 1);
+            main_grid.attach (power_combobox.label, 0, 3, 1, 1);
+            main_grid.attach (power_combobox, 1, 3, 1, 1);
 
             return main_grid;
         }
@@ -391,17 +379,6 @@ namespace Power {
 
             return false;
         }
-
-        private static void run_dpms_helper () {
-            try {
-                string[] argv = { "elementary-dpms-helper" };
-                Process.spawn_async (null, argv, Environ.get (),
-                    SpawnFlags.SEARCH_PATH | SpawnFlags.STDERR_TO_DEV_NULL | SpawnFlags.STDOUT_TO_DEV_NULL,
-                    null, null);
-            } catch (SpawnError e) {
-                warning ("Failed to reset dpms settings: %s", e.message);
-            }
-        }
     }
 }
 
