diff --git a/src/extensions/Makefile.am b/src/extensions/Makefile.am
index eb41f42..f66f0df 100644
--- a/src/extensions/Makefile.am
+++ b/src/extensions/Makefile.am
@@ -1,11 +1,11 @@
 SUBDIRS = test
-if HAVE_LIBSMACK
-if HAVE_LIBECRYPTFS
-SUBDIRS += tizen
-endif
-endif
-if HAVE_LIBECRYPTFS
-if HAVE_LIBTSPI
-SUBDIRS += ostro
-endif
-endif
+#if HAVE_LIBSMACK
+#if HAVE_LIBECRYPTFS
+#SUBDIRS += tizen
+#endif
+#endif
+#if HAVE_LIBECRYPTFS
+#if HAVE_LIBTSPI
+#SUBDIRS += ostro
+#endif
+#endif
