diff --git a/CMakeLists.txt b/CMakeLists.txt
index 817813a..322703d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,6 +36,8 @@ list (APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
 add_definitions (-w)
 add_definitions (-DGNOME_DESKTOP_USE_UNSTABLE_API)
 
+find_package (PkgConfig)
+
 # Configure file
 configure_file (${CMAKE_SOURCE_DIR}/config.vala.cmake ${CMAKE_BINARY_DIR}/src/config.vala)
 add_definitions (-DGETTEXT_PACKAGE=\"${GETTEXT_PACKAGE}\")
diff --git a/set-wallpaper-contract/CMakeLists.txt b/set-wallpaper-contract/CMakeLists.txt
index b096210..d0f4a64 100644
--- a/set-wallpaper-contract/CMakeLists.txt
+++ b/set-wallpaper-contract/CMakeLists.txt
@@ -5,7 +5,6 @@ set (DEPENDENCIES granite)
 
 configure_file (set-wallpaper.contract.in ${CMAKE_CURRENT_BINARY_DIR}/set-wallpaper.contract)
 
-find_package (PkgConfig)
 pkg_check_modules (DEPS REQUIRED ${DEPENDENCIES})
 
 vala_precompile (WALLPAPER_CONTRACT_VALA_C ${EXEC_NAME} ${SOURCES} PACKAGES ${DEPENDENCIES} posix)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f3c56b7..5b1b273 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -12,7 +12,6 @@ add_definitions (${DEPS_CFLAGS})
 link_libraries (${DEPS_LIBRARIES})
 link_directories (${DEPS_LIBRARY_DIRS})
 
-find_package (Vala REQUIRED)
 include (ValaVersion)
 ensure_vala_version ("0.22.0" MINIMUM)
 
